### DISCUSSION TITLE
명시적 타입변환과 암묵적 타입변환이 무엇인지 설명해주세요.

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#11] [타입변환과 단축 평가]
- 명시적 타입변환과 암묵적 타입변환이 무엇인지 설명해주세요.
  - 명시적 타입변환과 암묵적 타입변환 중, 어떤 방법이 좋다고 생각하나요? 
  - 생성자 함수를 사용하여 타입을 변환할 때 new 연산자 없이 호출하는 이유는 무엇인가요?

**Comment by publdaze** [comment]
> - 명시적 타입변환과 암묵적 타입변환이 무엇인지 설명해주세요.
- 명시적 타입 변환은
  - 개발자가 의도적으로 타입 변환
  - 빌트인 생성자 함수에 new 없이 호출하여 타입 변환 가능
  - toString, parseInt, parseFloat등 메소드로 타입 변환 가능
- 암묵적 타입 변환은
  - 자바스크립트 엔진에 의한 의도하지 않은 타입 변환
  - 문자열 연결 연산자 -> 문자로 타입 변환
  - 산술 연산자 -> 숫자로 타입 변환

>  - 명시적 타입변환과 암묵적 타입변환 중, 어떤 방법이 좋다고 생각하나요? 
- 가독성(간결함) 측면에서 암묵적 타입 변환이 선호되는 경우도 있지만 개인적으로는 명시적 타입 변환을 더 선호하는 편
  - 예측하지 못한 암묵적으로 타입 변환이 발생할 수 있어서 개발자가 유의해서 작성해야 함
  - 실수나 버그를 방지하기 위해 명시적 타입 변환 선호

>  - 생성자 함수를 사용하여 타입을 변환할 때 new 연산자 없이 호출하는 이유는 무엇인가요?
- new 연산자와 같이 호출하면 객체 타입으로 지정되기 때문에 원시 타입으로 변환하기 위해 new 연산자 없이 호출

**Comment by pipisebastian** [comment]
>명시적 타입변환과 암묵적 타입변환이 무엇인지 설명해주세요.

명시적 타입변환(타입 캐스팅) : 개발자가 의도적으로 값의 타입을 변환
암묵적 타입변환(타입강제변환) : 의도와 상관없이 자바스크립트 엔진에 의해 암묵적으로 타입이 자동 변환

>명시적 타입변환과 암묵적 타입변환 중, 어떤 방법이 좋다고 생각하나요?

상황에 따라 이해하기 암묵적 타입변환이 코드가 간결할 수 있겠지만, 실수를 방지하기 위해 명시적 타입변환을 사용하는 것이 좋다고 생각합니다.

>생성자 함수를 사용하여 타입을 변환할 때 new 연산자 없이 호출하는 이유는 무엇인가요?

new 로 만들 경우, 원시값이 아니라 객체를 반환하기 때문입니다.
